application-admintools-default/src/main/java/com/xwiki/admintools/internal/wikiUsage/UsageDataProvider.java

```diff
@@ -0,0 +1,213 @@
+    {
+        extensionsPingDataProvider.provideData(ping);
+        return ping.getExtensions().size();
+    }
+
+    /**
+     * Get the number of users in instance using {@link UsersPing}.
+     *
+     * @return the number of users in the XWiki instance.
+     */
+    public long getInstanceUsersCount()
+    {
+        usersPingDataProvider.provideData(ping);
+        return ping.getUsers().getTotal();
+    }
+
+    /**
+     * Retrieve info about the size of a given wiki.
+     *
+     * @param wikiDescriptor the wiki for which the data will be retrieved.
+     * @return a {@link WikiSizeResult} containing info about the size of the wiki.
+     * @throws QueryException if there are any exceptions while running the queries for data retrieval.
+     */
+    public WikiSizeResult getWikiSize(WikiDescriptor wikiDescriptor) throws QueryException
+    {
+        WikiSizeResult wikiData = new WikiSizeResult();
+        String wikiId = wikiDescriptor.getId();
+        wikiData.setName(wikiDescriptor.getPrettyName());
+        wikiData.setUserCount(getWikiUserCount(wikiId));
+        wikiData.setDocumentsCount(getWikiDocumentsCount(wikiId));
+        wikiData.setAttachmentsCount(getWikiAttachmentsCount(wikiId));
+        wikiData.setAttachmentsSize(readableSize(getWikiAttachmentSize(wikiId)));
+
+        return wikiData;
+    }
+
+    private Long getWikiUserCount(String wikiId) throws QueryException
+    {
+        Query query = this.queryManager.createQuery("SELECT COUNT(DISTINCT doc.fullName) FROM Document doc, "
+            + "doc.object(XWiki.XWikiUsers) AS obj WHERE doc.fullName NOT IN ("
+            + "SELECT doc.fullName FROM XWikiDocument doc, BaseObject objLimit, IntegerProperty propActive "
+            + "WHERE objLimit.name = doc.fullName AND propActive.id.id = objLimit.id AND propActive.id.name = 'active' "
+            + "AND propActive.value = 0)", Query.XWQL).setWiki(wikiId);
+        List<Long> results = query.execute();
+        return results.get(0);
+    }
+
+    private long getWikiDocumentsCount(String wikiId) throws QueryException
+    {
+        List<Long> results =
+            this.queryManager.createQuery("", Query.XWQL).setWiki(wikiId).addFilter(this.countFilter).execute();
+        return results.get(0);
+    }
+
+    private Long getWikiAttachmentSize(String wikiId) throws QueryException
+    {
+        List<Long> results = this.queryManager.createQuery(
+            "select sum(attach.longSize) from XWikiAttachment attach, XWikiDocument doc where attach.docId=doc.id",
+            Query.XWQL).setWiki(wikiId).execute();
+        return results.get(0);
+    }
+
+    private Long getWikiAttachmentsCount(String wikiId) throws QueryException
+    {
+        List<Long> results = this.queryManager.createQuery(
+                "select count(attach) from XWikiAttachment attach, XWikiDocument doc where attach.docId=doc.id",
+                Query.XWQL).setWiki(wikiId).addFilter(this.countFilter).execute();
```

Comment:
Why do you need to add the count filter if you call the ``count`` function yourself in the select clause?  Moreover, the count filter currently works only with document queries that include the document full name in the select clause. See https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-query/xwiki-platform-query-manager/src/main/java/org/xwiki/query/internal/CountDocumentFilter.java . So I'm not sure what's the effect of adding the count filter here.